### PR TITLE
fix:SingleSelect doesnot-close

### PR DESCRIPTION
**Pull request by monishshah-crest**

### CODE REVIEW

**Comment by monishshah-crest** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pytest_splunk_addon_ui_smartx/components/controls/single_select.py [hunk 1]
```diff
@@ -155,7 +155,10 @@ def get_single_value(self):
             self.select(selected_val, open_dropdown=False)
         elif self.searchable:
             self.input.send_keys(Keys.ESCAPE)
-        elif first_element:
+        else:
+            for each in self.get_elements('values'):
+                first_element = each
+                break
             self.select(first_element.text.strip(), open_dropdown=False)
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon_ui_smartx/components/controls/single_select.py:162]: Can you please update the code to: 
```
else:
     for each in self.get_elements('values'):
             self.select(each.text.strip(), open_dropdown=False)
             break            
```
To handle the scenario if the dropdown did not open and there are no option visible.

Comment (monishshah-crest) [reply pytest_splunk_addon_ui_smartx/components/controls/single_select.py]: Done

**Review by jayjoshi-crest — APPROVED** [review]

**Review by rajkarkar-crest — APPROVED** [review]

**Pull request merged**